net/dns/resolver/tsdns.go

```diff
@@ -71,7 +72,9 @@ type Config struct {
 	// Queries only match the most specific suffix.
 	// To register a "default route", add an entry for ".".
 	Routes map[dnsname.FQDN][]*dnstype.Resolver
-	// LocalHosts is a map of FQDNs to corresponding IPs.
+	// Hosts maps either domain names or wildcards to IP(s).
+	// If the map key begins with a dot (e.g. ".foo.com.") then it's considered a wildcard,
+	// matching *.foo.com.
```

Comment:
would be nice to explicitly say whether or not foo.com is matched by .foo.com here.